File: bank-service/src/main/java/rs/edu/raf/repository/CardRepository.java
package rs.edu.raf.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import rs.edu.raf.model.card.Card;

import java.util.List;
import java.util.Optional;

@Repository
public interface CardRepository extends JpaRepository<Card, Long> {
    Optional<Card> findByNumber(String number);
    List<Card> findByBankAccount_Owner(Long owner);
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/repository/CardRepository.kt
package org.example.bankservicekotlin.repository

import org.example.bankservicekotlin.model.card.Card
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface CardRepository:JpaRepository<Card,Long> {
    fun findByNumber(number: String): Optional<Card>
    fun findByBankAccount_Owner(owner: Long): List<Card>
}